packages/types/src/index.ts

```diff
@@ -27,6 +28,7 @@ export interface Actions {
   startActivation: () => () => void
   update: (stateUpdate: Web3ReactStateUpdate) => void
   resetState: () => void
+  setError: (error: ProviderRpcError) => void
```

Comment:
Why can't you set error as part of `update`? This is symmetrical with the way you would read an error - out of the `Web3ReactState`.

    export type Web3ReactState =
      | ...
      | {
          error: ProviderRpcError
      }

Using this pattern, I would expect calling `update({error})` to set an error, and _maybe_ to also reset state.

Reply:
However this ends up should be documented using comments or doc comments (whichever is already in use) so that it doesn't need to be re-divined from the code at a later date (when docs are written 🤞)

Reply:
sounds good done. used update instead.